app/cloud_tasks/cloud_task_publishers.py

```diff
@@ -0,0 +1,57 @@
+from google import auth
+from google.cloud.tasks_v2 import CloudTasksClient, HttpMethod
+from google.cloud.tasks_v2.types.task import Task
+from structlog import get_logger
+
+from app.settings import SUBMISSION_CONFIRMATION_CONSUMER
+
+from .exceptions import CloudTaskCreationFailed
+
+logger = get_logger(__name__)
+
+
+class CloudTaskPublisher:
+    def __init__(self):
+        self._client = CloudTasksClient()
+
+        _, self._project_id = auth.default()
+
+        url = f"https://europe-west2-{self._project_id}.cloudfunctions.net/{SUBMISSION_CONFIRMATION_CONSUMER}"
+        self._task = {
+            "http_request": {
+                "http_method": HttpMethod.POST,
+                "url": url,
+                "oidc_token": {
+                    "service_account_email": f"cloud-function-invoker@{self._project_id}.iam.gserviceaccount.com"
+                },
+                "headers": {
+                    "Content-type": "application/json",
+                },
+            },
+        }
+
+    def _create(self, payload: bytes, queue_name: str) -> Task:
+        self._parent = self._client.queue_path(
+            self._project_id, "europe-west2", queue_name
+        )
+
+        logger.info("creating cloud task")
+        task = self._task.copy()
+        task["http_request"]["body"] = payload
+        return self._client.create_task(request={"parent": self._parent, "task": task})
+
+    def create_task(self, payload: bytes, queue_name: str) -> None:
+        try:
+            self._create(payload, queue_name)
+            logger.info("task published successfully")  # pragma: no cover
```

Comment:
The positive log message should align with the exception message:
```suggestion
            logger.info("task created successfully")  # pragma: no cover
```